pkg/render/kube-controllers.go

```diff
@@ -54,55 +57,55 @@ func KubeControllers(
 	authentication *operator.Authentication,
 	enabledESOIDCWorkaround bool,
 	clusterDomain string,
-	esAdminSecret *v1.Secret,
+	kubeControllersGatewaySecret *v1.Secret,
 	metricsPort int,
 ) *kubeControllersComponent {
-	var elasticsearchUserSecret *v1.Secret
-	if esAdminSecret != nil {
-		elasticsearchUserSecret = &v1.Secret{
+	if kubeControllersGatewaySecret == nil {
```

Comment:
I made a comment in the logstorage controller about how this shouldn't be created here. This also introduces an problem for the operator when used in OS, where there is no Elasticsearch (i.e. don't created create elasticsearch secrets if there's no elasticsearch).